docs/source/cuda/cuda_array_interface.rst

```diff
@@ -69,31 +69,307 @@ The following are optional entries:
+       ``stream`` entry is set to ``3``.
+
+* If there is no work enqueued on the data, then the ``stream`` entry may be
+  either ``None``, or not provided.
+
+Optionally, to facilitate the User to relax the conformance to synchronization
+semantics:
+
+* Producers may provide a configuration option to always set ``stream`` to
+  ``None``.
+* Consumers may provide a configuration option to ignore the value of ``stream``
+  and act as if it were ``None`` or not provided.  This elides synchronization
+  on the Producer-provided streams, and allows enqueuing work on streams other
+  than that provided by the Producer.
+
+These options should be not be set by default in either a Producer or a
+Consumer. The exact mechanism by which these options are set, and related
+options that Producers or Consumers might provide to allow the user further
+control over synchronization behavior are not prescribed by the CAI
+specification.
+
+
+Synchronization in Numba
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Numba is neither strictly a Producer nor a Consumer - it may be used to
+implement either by a User. In order to facilitate the correct implementation of
+synchronization semantics, Numba exhibits the following behaviors related to
+synchronization of the interface:
+
+- When Numba acts as a Consumer (for example when an array-like object is passed
+  to a kernel launch): If ``stream`` is an integer, then Numba will immediately
+  synchronize on the provided ``stream``, and set the *default stream* of the
+  array to the provided stream.
+- When Numba acts as a Producer (when the ``__cuda_array_interface__`` property
+  of a Numba Device Array is accessed): If the exported Device Array has a
+  *default stream*, then it is given as the ``stream`` entry. Otherwise,
+  ``stream`` is set to ``None``.
+
+.. note:: In Numba's terminology, the *default stream* for a Device Array is a
+          property of the Device Array specifying the stream in which Numba will
+          enqueue asynchronous transfers on if no other stream is provided as an
+          argument to the function invoking the transfer. It is not the same as
+          the `Default Stream
+          <https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#default-stream>`_
+          in normal CUDA terminology.
+
+Resulting from these properties, these consequences are intended:
+
+- Exchanging data either as a Producer or a Consumer should generally be
+  correct, provided the other side of the interaction also follows the CAI
+  synchronization semantics.
+- The User is expected not to launch kernels or other operations on streams that
+  are not the default stream for their parameters, because doing so would
+  violate the requirements for a Producer.
+
+  - Warning the user when they do this could be added. The present
+    implementation is a minimal prototype to help illustrate the interface
+    specification.
+
+The user may override synchronization behaviour in Numba by setting the
```

Comment:
```suggestion
The User may override synchronization behaviour in Numba by setting the
```

Also I note British spelling of behaviour here, American earlier. Not a problem for an international project, IMO.

Reply:
Thanks for spotting the spelling - I've replaced it with the US version, as I try to consistently use US spellings in Numba.